src/webapps/azext_webapps/custom.py

```diff
@@ -0,0 +1,144 @@
+    full_sku = _get_sku_name(sku)
+
+    if location is None:
+        locs = client.list_geo_regions(sku, True)
+        available_locs = []
+        for loc in locs:
+            available_locs.append(loc.geo_region_name)
+        location = available_locs[0]
+
+    # Remove spaces from the location string, incase the GeoRegion string is used
+        loc_name = location.replace(" ", "")
+
+    asp = "appsvc_asp_{}_{}".format(os_val, loc_name)
+    rg = "appsvc_rg_{}_{}".format(os_val, loc_name)
+
+    # the code to deploy is expected to be the current directory the command is running from
+    src_dir = os.getcwd()
+
+    # if dir is empty, show a message in dry run
+    do_deployment = False if os.listdir(src_dir) == [] else True
+    package_json_path = is_node_application(src_dir)
+
+    str_no_contents_warn = ""
+    if not do_deployment:
+        str_no_contents_warn = "[Empty directory, no deployment will be triggered]"
+
+    if package_json_path == '':
+        node_version = "[No package.json file found in root directory, not a Node app?]"
+        version_used_create = "8.0"
+    else:
+        with open(package_json_path) as data_file:
+            data = json.load(data_file)
+            node_version = data['version']
+            version_used_create = get_node_runtime_version_toSet()
+
+    # ResourceGroup: check if default RG is set
+    default_rg = cmd.cli_ctx.config.get('defaults', 'group')
+    if (default_rg and check_resource_group_supports_linux(cmd, default_rg, location)):
+        rg = default_rg
+        rg_mssg = "[Using default ResourceGroup]"
+    else:
+        rg_mssg = ""
+
+    runtime_version = "{}|{}".format(language, version_used_create)
+    src_path = "{} {}".format(src_dir.replace("\\", "\\\\"), str_no_contents_warn)
+    rg_str = "{} {}".format(rg, rg_mssg)
+
+    dry_run_str = r""" {
+            "name" : "%s",
+            "appServicePlan" : "%s",
+            "resourceGroup" : "%s",
+            "sku": "%s",
+            "os": "%s",
+            "location" : "%s",
+            "src_path" : "%s",
+            "version_detected": "%s",
+            "version_to_create": "%s"
+            }
+            """ % (name, asp, rg_str, full_sku, os_val, location, src_path,
+                   node_version, runtime_version)
+
+    create_json = json.dumps(json.loads(dry_run_str), indent=4, sort_keys=True)
+    if dryrun:
+        logger.warning("""
+            Web app will be created with the below configuration,
+            re-run command without the --dryrun flag to create & deploy a new app
+            """)
+        return logger.warning(create_json)
+
+    # create RG if the RG doesn't already exist
+    if not check_resource_group_exists(cmd, rg):
+        logger.warning("Creating ResourceGroup '%s' ...", rg)
+        create_resource_group(cmd, rg, location)
+        logger.warning("ResourceGroup creation complete")
+    else:
+        logger.warning("Resource Group '%s' already exists.", rg)
```

Comment:
Would suggest `Resource group` instead of `ResourceGroup` and `Resource Group` in the log messages.